### PR TITLE
Backport of generator parameter scan functionality to 71x

### PR DESCRIPTION
**Pull request by bendavid**
Also brings Pythia8Interface ~in sync with master in general for easier backporting in the future.

@mkirsano

### CODE REVIEW

**Comment by bendavid** [comment]
@intrepid42  in order to bring things into sync, I effectively reverted your pull requests related to recent powheg veto hooks and then re-backported them from master.  Please check that the end result makes sense.

**Comment by perrozzi** [comment]
please test

File: GeneratorInterface/Pythia8Interface/plugins/PowhegResHook.cc [hunk 1]
```diff
@@ -57,9 +57,9 @@ double PowhegResHook::scaleResonance( const int iRes, const Event& event) {
     
     for (int i = 0; i < nDau; i++) {
       int iDau = event[iRes].daughterList()[i];
-      if      (event[iDau].id() == 21) idg = iDau;
-      else if (event[iDau].id()  >  0) idq = iDau;
-      else if (event[iDau].id()  <  0) ida = iDau;
+      if      (abs(event[iDau].id()) == 21) idg = iDau;
+      else if (abs(event[iDau].id())  >  0) idq = iDau;
+      else if (abs(event[iDau].id())  <  0) ida = iDau;
```
Comment (intrepid42) [inline GeneratorInterface/Pythia8Interface/plugins/PowhegResHook.cc]: This introduces a bug (abs() is always > 0)

File: GeneratorInterface/Pythia8Interface/test/pythia8ex12_cfg.py [hunk 2]
```diff
@@ -1,68 +0,0 @@
-import FWCore.ParameterSet.Config as cms
```
Comment (intrepid42) [inline GeneratorInterface/Pythia8Interface/test/pythia8ex12_cfg.py]: Please add back the example config.

**Review by mseidel42 — CHANGES_REQUESTED** [review]
Ok, I have just 2 small comments

File: GeneratorInterface/Pythia8Interface/plugins/PowhegResHook.cc (hunk 1 above)
Comment (bendavid) [inline GeneratorInterface/Pythia8Interface/plugins/PowhegResHook.cc]: Ok thanks, indeed I missed a commit when restoring your changes, will update the pull request accordingly.

File: GeneratorInterface/Pythia8Interface/test/pythia8ex12_cfg.py (hunk 2 above)
Comment (bendavid) [inline GeneratorInterface/Pythia8Interface/test/pythia8ex12_cfg.py]: Will do.

**Comment by bendavid** [comment]
This still needs updates before it can be merged, given the comments from @intrepid42

**Comment by bendavid** [comment]
Alright, I think this should be ok now.

**Comment by perrozzi** [comment]
please test

**Comment by perrozzi** [comment]
@efeyazgan

**Comment by perrozzi** [comment]
@efeyazgan

**Comment by perrozzi** [comment]
from what I can see and I learnt by discussing with Josh, things look ok and the feature will be active only when explicitly set ON, approving from the GEN side

**Comment by perrozzi** [comment]
+1

**Comment by bendavid** [comment]
One caveat which I previously forgot about.
This pull request does add the new GenLumiInfoHeader data product as a lumi product even for workflows which don't use the new functionality.

This might cause problems for future 2015 MC requests with DIGI-RECO in 76x unless the corresponding data formats are added in 76x as well.  (Everything is already present in 80x and above, so no problem for 2016 MC)

**Comment by perrozzi** [comment]
I have not seen a single request for 2015 MC in the last 18 months :-)

**Comment by davidlange6** [comment]
looks like its been at least 9 months since a 2015 request. Lets agree that 2015 GEN-SIM requests are now frozen to use the existing 71x releases.

**Comment by davidlange6** [comment]
+1

**Comment by perrozzi** [comment]
agreed, thanks a lot

**Pull request merged**